### PR TITLE
fix: moves the kafkaProducer instantiation higher in the hierarchy

### PR DESCRIPTION
**Pull request by opaolini**
also removes it from the top level in swap_handlers.ts

# Description

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/services/swap_service.ts [hunk 1]
```diff
@@ -197,9 +200,13 @@ export class SwapService {
         firmQuoteValidator?: RfqFirmQuoteValidator | undefined,
         rfqDynamicBlacklist?: RfqDynamicBlacklist,
         private readonly _pairsManager?: PairsManager,
+        _kafkaClient?: Kafka,
     ) {
         this._provider = provider;
         this._firmQuoteValidator = firmQuoteValidator;
+        this._kafkaClient = _kafkaClient;
+        // We initialize this async via connectKafka()
```
Comment (phil-ociraptor) [inline src/services/swap_service.ts:208]:
```suggestion
        // We initialize this async via connectKafkaOrThrow()
```

File: src/services/swap_service.ts [hunk 2]
```diff
@@ -629,6 +636,15 @@ export class SwapService {
         };
     }
 
+    public async connectKafkaOrThrow(): Promise<void> {
```
Comment (phil-ociraptor) [inline src/services/swap_service.ts:639]: Linter will complain

```suggestion
    public async connectKafkaOrThrowAsync(): Promise<void> {
```

File: src/services/swap_service.ts [hunk 3]
```diff
@@ -629,6 +636,15 @@ export class SwapService {
...
+        if (!this._kafkaClient) {
+            throw Error('Kafka client was not passed in.');
+        }
+        const producer = this._kafkaClient.producer();
+        await producer.connect();
+        this.kafkaProducer = producer;
+    }
```
Comment (kimpers) [inline src/services/swap_service.ts:646]: Previously Kafka was optional, to run 0x API locally you didn't need to connect it. Is there any advantage to making it required, or could we keep it optional but still error out if it should connect but fails?

**Review by kimpers — APPROVED** [review]
LGTM 👍

**Pull request closed without merging**